lib/Common/CommonDefines.h

```diff
@@ -134,22 +137,32 @@
 #endif
 
 // dep: IDebugDocumentContext
+#if !BUILD_WITHOUT_SCRIPT_DEBUG
 #define ENABLE_SCRIPT_DEBUGGING
+#endif
 
 // GC features
 #define BUCKETIZE_MEDIUM_ALLOCATIONS 1              // *** TODO: Won't build if disabled currently
 #define SMALLBLOCK_MEDIUM_ALLOC 1                   // *** TODO: Won't build if disabled currently
 #define LARGEHEAPBLOCK_ENCODING 1                   // Large heap block metadata encoding
+#ifndef CHAKRACORE_LITE
 #define IDLE_DECOMMIT_ENABLED 1                     // Idle Decommit
-#define RECYCLER_PAGE_HEAP                          // PageHeap support
+#endif
+
+#if defined(NTBUILD) || defined(ENABLE_DEBUG_CONFIG_OPTIONS)
+#define RECYCLER_PAGE_HEAP                          // PageHeap support, on by default, off in ChakraCore release build
+#endif
+
 
 #ifdef _WIN32
 #define SYSINFO_IMAGE_BASE_AVAILABLE 1
 #define ENABLE_CONCURRENT_GC 1
 #define ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP 1 // Only takes effect when ENABLE_CONCURRENT_GC is enabled.
 #define ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP_USE_SLIST 1 // Use Interlocked SLIST for allocableHeapBlockList
 #define SUPPORT_WIN32_SLIST 1
+#ifndef CHAKRACORE_LITE
 #define ENABLE_JS_ETW                               // ETW support
```

Comment:
I would like the "lite" mode to be as bare bone as possible.  
It is easy for the user to reenable these piece meal
It would also allow us to test builds with these feature disabled as well.